SwiftStyleGuide.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  SwiftStyleGuide.swift
+//  PurchasesCoreSwift
+//
+//  Created by Andrés Boedo on 7/12/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+protocol ValuePrintable {
+    // make protocol variables read-only unless needed
+    func foo() -> String
+}
+
+// use protocol extensions for default implementations of protocols for types
+extension ValuePrintable where Self: NSNumber {
+    func foo() -> String {
+        return "the number as float is: \(self.floatValue)"
+    }
+}
+
+// prefer structs to classes whenever possible.
+// keep in mind that structs are value types.
+// More info here: https://developer.apple.com/documentation/swift/choosing_between_structures_and_classes
+struct MyStruct {
+
+    // don't explicitly define types unless needed.
+    // prefer let to var whenever possible.
+
+    // public properties, then internal, then private.
+    public let mercury = false
```

Comment:
+100, same for types. I'll update

Reply:
added this [swiftlint check](https://realm.github.io/SwiftLint/missing_docs.html)

Reply:
aaand after adding it I realized that it'll be super annoying to have as a rule during the migration since we're going to have a ton of stuff public that we'll revert, so I'm going to comment it out with a todo to re-enable after the migration is done

Reply:
HA! Good idea